### PR TITLE
allow `repr(align = x)` on inherent methods

### PR DESCRIPTION
**Pull request by fee1-dead**
Discussion: https://github.com/rust-lang/rust/issues/82232#issuecomment-905929314

### CODE REVIEW

File: tests/codegen/align-fn.rs [hunk 1]
```diff
@@ -7,3 +7,12 @@
...
 pub fn fn_align() {}
+
+pub struct A;
+
+impl A {
+    // CHECK: align 16
+    #[no_mangle]
+    #[repr(align(16))]
+    pub fn method_align(self) {}
+}
```
Comment (WaffleLapkin) [inline tests/codegen/align-fn.rs:18]: Can you also add a check for trait methods and associated functions, just to be sure? i.e.
```suggestion

    // CHECK: align 16
    #[no_mangle]
    #[repr(align(16))]
    pub fn associated_fn() {}
}

trait T {
    fn trait_fn() {}
    fn trait_method(self) {}
}

impl T for A {
    // CHECK: align 16
    #[no_mangle]
    #[repr(align(16))]
    fn trait_fn() {}
    
    // CHECK: align 16
    #[no_mangle]
    #[repr(align(16))]
    fn trait_method(self) {}
}
```

Comment (fee1-dead) [reply tests/codegen/align-fn.rs:18]: Should those be allowed? The current change doesn't allow those, but I can change it to include trait impls as well.

Comment (WaffleLapkin) [reply tests/codegen/align-fn.rs:18]: I think yes, they should? Don't really see a reason to not allow these (btw maybe a test for default bodies should be added too, although I'm not sure if it's easy to do...).

**Review by WaffleLapkin — APPROVED** [review]

**Comment by WaffleLapkin** [comment]
Do we not have a test which shows the `passes_attr_application_struct_enum_function_method_union` error message?... @fee1-dead would you like to add a test for that?

@bors r+ rollup

**Pull request merged**